### DISCUSSION TITLE
Running tts on long text

### DISCUSSION THREAD

**Discussion by dln22** [NVIDIA-NeMo/NeMo#5152] [Q&A]
I am facing an out of memory issue when running the tts model on long sentences. I worked around it by splitting the text and processing it in chunks and then combining produced audio files. This, however, produces small glitches in the final audio, in places where the audio chunks were merged. I wonder if there is any other way to convert long text to audio in one go, without running out of memory?

Models I am using:
  spectrogram_model="tts_en_fastpitch"
  vocoder_model="tts_hifigan"

**Comment by XuesongYang** [comment]
It would be better to segment your input text based on punctuation and keep each segment around 15 seconds.
